pages/OAuth2Server/OAuthProvider.md

```diff
@@ -0,0 +1,40 @@
+---
+title: OAuthProvider
+permalink: /oauth2server/oauthprovider
+---
+
+# OAuthProvider
+
+The `OAuthProvider` class is an interface for the dependency injection definition:
+
+```typescript
+// TypeScript
+export interface OAuthProvider {
+    clientModel(): OAuth2BaseModel;
+    userModel(getUser: GetOAuthUserInteractor, loginUser: LoginOAuthUserInteractor): OAuth2UserModel;
+}
+```
+
+It exposes two methods, one for obtaining a [`OAuth2BaseModel`](OAuth2ServerModels.md) and a [`OAuth2UserModel`](OAuth2ServerModels.md).
+
+Currently, there is only one implementation of that provider, described below.
```

Comment:
I would say that we can add that you can implement your own implementations of the Provider and use it if you needed

Reply:
added